### ISSUE TITLE
Remote-chunking implementation does not support parallel executions of the same job [BATCH-2228]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-batch#1372]
**[Tomas Klubal](https://jira.spring.io/secure/ViewProfile.jspa?name=blondak)** opened **[BATCH-2228](https://jira.spring.io/browse/BATCH-2228?redirect=false)** and commented

Current implementation of the remote-chunking does not support same job instance executing in parallel (2 different job executions). For example running same job with different parameters at the same time. These two job executions are stealing each others reply messages.

---

**Affects:** 3.0.0

**Issue Links:**
- [BATCH-2507](https://jira.spring.io/browse/BATCH-2507) ChunkMessageChannelItemWriter Reply Channel Issues
 (_**"is duplicated by"**_)
- [BATCH-2508](https://jira.spring.io/browse/BATCH-2508) ChunkMessageChannelItemWriter Reply Channel Issues
 (_**"is duplicated by"**_)

**Comment by akashsharma3030** [comment]
Is there any solution for this problem of running parallel execution of spring remote chunking job

**Comment by fmbenhassine** [comment]
Closing this as duplicate of #1372.

**Comment by fmbenhassine** [comment]
@akashsharma3030 

> Is there any solution for this problem of running parallel execution of spring remote chunking job

Similar to the `MessageChannelPartitionHandler` that should be step-scoped or job-scoped to support parallel executions (see [the note in its Javadoc](https://docs.spring.io/spring-batch/docs/current/api/org/springframework/batch/integration/partition/MessageChannelPartitionHandler.html)),  Have you tried to make the `ChunkMessageChannelItemWriter` step-scoped?

I think the lack of a minimal example is what is missing to address this issue and fix it correctly. So if you have a reproducer, it would be of a great help to share it with us.

**Comment by Vanguard-Fx** [comment]
@fmbenhassine i think this issue is related to this thread:
https://stackoverflow.com/a/37334833/16742414

My business logic needs to run more than one job in parallel, but using the same channel they collide

I've created an example project with this problem here: https://github.com/Vanguard-Fx/SpringBatchRemoteChunking

**Comment by fmbenhassine** [comment]
I wonder if this is similar to #4133, where we need to use a `SubscribableChannel` instead of a `PollableChannel` in `ChunkMessageChannelItemWriter`.

Planning this for 6.1 to include the same fix in both classes at the same time.
